pkg/kubelet/rkt/rkt.go

```diff
@@ -1521,16 +1523,215 @@ func (r *Runtime) RemoveImage(image kubecontainer.ImageSpec) error {
+	}
+
+	var latestPod *rktapi.Pod
+	var latestPodManifest *appcschema.PodManifest
+	var latestRestartCount int = -1
+	var latestCreationTime time.Time
+
+	// Only use the latest pod to get the information, (which has the
+	// largest restart count).
+	for _, rktpod := range listResp.Pods {
+		inspectReq := &rktapi.InspectPodRequest{rktpod.Id}
+		inspectResp, err := r.apisvc.InspectPod(context.Background(), inspectReq)
+		if err != nil {
+			glog.Warningf("rkt: Couldn't inspect rkt pod, (uuid %q): %v", rktpod.Id, err)
+			continue
+		}
+
+		var manifest appcschema.PodManifest
+		if err = json.Unmarshal(inspectResp.Pod.Manifest, &manifest); err != nil {
+			glog.Warningf("rkt: Couldn't unmarshal pod manifest for rkt pod, (uuid %q): %v", rktpod.Id, err)
+			continue
+		}
+
+		creationTimeStr, ok := manifest.Annotations.Get(k8sRktCreationTimeAnno)
+		if !ok {
+			glog.Warningf("rkt: No creation timestamp in pod manifest for rkt pod, (uuid %q)", rktpod.Id)
+			continue
+		}
+
+		unixSec, err := strconv.ParseInt(creationTimeStr, 10, 64)
+		if err != nil {
+			glog.Warningf("rkt: Invalid creation timestamp in pod manifest for rkt pod, (uuid %q)", rktpod.Id)
+			continue
+		}
+
+		if countString, ok := manifest.Annotations.Get(k8sRktRestartCountAnno); ok {
+			num, err := strconv.Atoi(countString)
+			if err != nil {
+				glog.Warningf("rkt: Unrecognized restart count value (%q) for rkt pod, (uuid %q)", countString, rktpod.Id)
+				continue
+			}
+			if num > latestRestartCount {
+				latestRestartCount = num
+				latestPod = inspectResp.Pod
+				latestPodManifest = &manifest
+				latestCreationTime = time.Unix(unixSec, 0)
+			}
+		}
+	}
+
+	// Cannot find any rkt pod associated with the api pod UID.
+	if latestPod == nil {
+		return nil, nil
+	}
+
+	// Try to fill the IP info.
+	for _, n := range latestPod.Networks {
+		if n.Name == defaultNetworkName {
+			podStatus.IP = n.Ipv4
+		}
+	}
+
+	for i, app := range latestPod.Apps {
+		// The order of the apps is determined by the rkt pod manifest.
+		// TODO(yifan): Let the server to unmarshal the annotations?
+		hashStr, ok := latestPodManifest.Apps[i].Annotations.Get(k8sRktContainerHashAnno)
+		if !ok {
+			glog.Warningf("rkt: No container hash in pod manifest for rkt pod, (uuid %q, app %q)", latestPod.Id, app.Name)
+			continue
+		}
+
+		hashNum, err := strconv.ParseUint(hashStr, 10, 64)
+		if err != nil {
+			glog.Warningf("rkt: Invalid hash value for rkt pod, (uuid %q, app %q)", latestPod.Id, app.Name)
+			continue
+		}
+
+		cs := kubecontainer.ContainerStatus{
+			ID:    buildContainerID(&containerID{uuid: latestPod.Id, appName: app.Name}),
+			Name:  app.Name,
+			State: appStateToContainerState(app.State),
+			// TODO(yifan): Use the creation/start/finished timestamp when it's implemented.
+			CreatedAt: latestCreationTime,
+			StartedAt: latestCreationTime,
+			ExitCode:  int(app.ExitCode),
+			Image:     app.Image.Name,
+			ImageID:   app.Image.Id,
```

Comment:
@yifan-gu @yujuhong In docker, the image id is added with a prefix "docker://" by kubelet. What is the image id like in rkt? Should we also add prefix to show image type? :)